File: wiki_java/src/main/java/com/github/klee0kai/wiki/init/AppComponent.java
package com.github.klee0kai.wiki.init;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.component.Init;

@Component
public interface AppComponent {

    FeatureModule feature();

    StarsDependencies starsDependencies();

    StarsDependencies2 starsDependencies2();

    @Init
    void initFeatureModule(FeatureModule featureModule);

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/init/AppComponent.kt
package com.github.klee0kai.wiki.init

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.component.Init

@Component
interface AppComponent {
    fun feature(): FeatureModule

    fun starsDependencies(): StarsDependencies

    fun starsDependencies2(): StarsDependencies2

    @Init
    fun initFeatureModule(featureModule: FeatureModule?)
}
